portal-impl/src/com/liferay/portlet/PortalPreferencesImpl.java

```diff
@@ -393,8 +396,19 @@ public int size() {
 
 	public void store() throws IOException {
 		try {
-			PortalPreferencesLocalServiceUtil.updatePreferences(
-				getOwnerId(), getOwnerType(), this);
+			if (!DBPartitionUtil.isPartitionEnabled() ||
+				((getOwnerId() != PortletKeys.PREFS_OWNER_ID_DEFAULT) &&
+				 (getOwnerType() == PortletKeys.PREFS_OWNER_TYPE_COMPANY))) {
+
+				PortalPreferencesLocalServiceUtil.updatePreferences(
+					getOwnerId(), getOwnerType(), this);
+
+				return;
+			}
+
+			CompanyLocalServiceUtil.forEachCompany(
+				company -> PortalPreferencesLocalServiceUtil.updatePreferences(
+					getOwnerId(), getOwnerType(), this));
```

Comment:
@tototrinh 
What does "this" refer to? the current instance or the company ?
Thank you.